.claude/settings.json

```diff
@@ -0,0 +1,38 @@
+{
+  "instructions": "You are working on Ops, the official Python framework for Juju charm development. This is a mature, production framework requiring careful attention to backward compatibility, type safety, and comprehensive testing. See AGENTS.md for detailed guidelines.",
+  "enableHooks": true,
+  "defaultModel": "sonnet",
+  "agentModels": {
+    "test-runner": "haiku",
+    "Explore": "haiku"
+  },
+  "memorySettings": {
+    "contextWindow": "large"
+  },
+  "projectContext": {
+    "name": "Ops",
+    "type": "Python Framework",
+    "pythonVersion": "3.10+",
+    "testCommand": "tox",
+    "formatCommand": "tox -e format",
+    "lintCommand": "tox -e lint"
+  },
+  "codeStyle": {
+    "linter": "ruff",
+    "typeChecker": "pyright",
+    "docstringStyle": "google",
+    "formatter": "ruff"
+  },
+  "permissions": {
+    "allow": [
+      "WebFetch(domain:documentation.ubuntu.com)",
+      "WebFetch(domain:docs.zizmor.sh)",
+      "WebFetch(domain:charmhub.io)",
+      "WebFetch(domain:github.com)",
```

Comment:
There's a bit of risk here, in that both of these allow anyone to add content. That means that there could be hidden instructions on these sites, and GitHub at least (maybe also Charmhub) can be used for exfiltration. I feel that for our use, this should be safe enough, and that you shouldn't let the agent have access to any data you wouldn't want to leave your system anyway.

However, if we want safer defaults, we could drop these and individually approve them as needed.